mechanicalsoup/stateful_browser.py

```diff
@@ -230,16 +230,32 @@ def links(self, url_regex=None, link_text=None, *args, **kwargs):
                          if a.text == link_text]
         return all_links
 
-    def find_link(self, *args, **kwargs):
+    def find_link(self, link=None, *args, **kwargs):
         """Find and return a link, as a bs4.element.Tag object.
 
         The search can be refined by specifying any argument that is accepted
         by :func:`links`. If several links match, return the first one found.
 
+        As a special-case, the first argument (``link``) has an
+        ``href`` attribute, then this element is returned. This is
+        simplifies the job of callers like :func:`follow_link`
+        and :func:`download_link` to allow passing a ``<a href=...>``
+        tag object to these methods. Otherwise, if this argument is
+        not ``None``, it is passed to :func:`links` and will be
+        considered as the ``url_regex`` argument.
+
         If no link is found, raise :class:`LinkNotFoundError`.
         """
+        if hasattr(link, 'attrs') and 'href' in link.attrs:
+            return link
+        if link is not None:
+            args = (link,) + args
         links = self.links(*args, **kwargs)
         if len(links) == 0:
+            if self.get_debug():
```

Comment:
No change requested here. Just wanted to say that I like the idea of moving this debug conditional from `follow_link` to `find_link`.